staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1512,3 +1517,68 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Keep temporary files. Like log files. Helpful for debugging.
+    bool keep_temporary_files = 1;
+    // Checkpoint/Restore the container with established TCP connections.
+    bool keep_network_state = 2;
+    // Location of the checkpoint archive used for export/import
+    string archive_location = 3;
+    // Instead of using a local checkpoint archive use a OCI image
+    // to store the checkpoint.
+    bool archive_is_oci_image = 4;
+    // Specify the checkpoint archive compression algorithm
+    // See github.com/containers/storage/pkg/archive/archive.go for
+    // possible values.
+    int64 compression = 5;
+}
+
+message CheckpointContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // Leave the container running after checkpointing
+    bool leave_running = 2;
+}
+
+message CheckpointContainerRequest {
+    // ID of the container to be checkpointed.
+    string container_id = 1;
+    // Options for checkpointing.
+    CheckpointContainerOptions options = 2;
+}
+
+message CheckpointContainerResponse {
+}
+
+message RestoreContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // ID of the sandbox into which the container should be restored
+    // If this is not specified the container will be restored
+    // in the sandbox it was checkpointed from.
+    string pod_sandbox_id = 2;
+    // Restoring multiple copies of a checkpointed container may
+    // require a new name for the restored container.
+    string name = 3;
+    // The labels and annotations are in the same form as in ContainerConfig
+    // during CreateContainerRequest.
+    map<string, string> labels = 4;
+    map<string, string> annotations = 5;
```

Comment:
Same question for this.

Reply:
Not needed. That was a left-over from the end-to-end implementation of the drain use case from the beginning of the discussions.